code/modules/maptext_alerts/location_blurbs.dm

```diff
@@ -0,0 +1,154 @@
+/obj/screen/text/blurb
+	maptext_height = 64
+	maptext_width = 512
+	screen_loc = "LEFT+1,BOTTOM+2"
+	/// Font size in pixels
+	var/font_size = 11
+	/// Font family
+	var/font_family = "Courier New"
+	/// Where text is aligned
+	var/text_alignment = "left"
+	/// Color of text in RGB
+	var/text_color = COLOR_WHITE
+	/// Color of text outline
+	var/text_outline_color = COLOR_BLACK
+	/// Width of text outline in pixels
+	var/text_outline_width = 1
+	/// Text that will be shown in blurb
+	var/blurb_text = ""
+	/// Number of chars from the `text` that will be displayed per interval. Defaults to 1
+	var/chars_per_interval = 1
+	/// The interval between chars rendering
+	var/interval = 1 DECISECONDS
+	/// Amount of time the blurb will be present on the screen. 0 means that blurb will dissappear immediately
+	var/hold_for = 0
+	/// Amount of time the blurbs appering (alpha changing from 0 to 255). 0 means blurb is fully opaque from the start
+	var/appear_animation_duration = 0
+	/// Amount of time the blurb takes to fade (alpha changing from 255 to 0). 0 means blurb is instantly removed from the screen after finished
+	var/fade_animation_duration = 0
+	// Colours of the background
+	var/background_r = 0
+	var/background_g = 0
+	var/background_b = 0
+	var/background_a = 0
+
+
+/obj/screen/text/blurb/proc/show_to(list/client/viewers)
+	if(!blurb_text || !viewers)
+		return
+
+	if(islist(viewers))
+		if(!length(viewers))
+			return
+
+	else
+		viewers = list(viewers)
+
+	for(var/client/viewer as anything in viewers)
+		viewer.screen += src
+
+	appear()
+	print_text()
+
+	if(hold_for)
+		addtimer(CALLBACK(src, TYPE_PROC_REF(/obj/screen/text/blurb, hide_from), viewers), hold_for)
+	else
+		hide_from(viewers)
+
+
+/obj/screen/text/blurb/proc/get_text_style()
+	PRIVATE_PROC(TRUE)
+
+	return {"\
+		font-family: [font_family], [initial(font_family)]; \
+		-dm-text-outline: [text_outline_width] [text_outline_color]; \
+		background-color: rgba([background_r], [background_g], [background_b], [background_a]); \
+		font-size: [font_size]px; \
+		text-align: [text_alignment]; \
+		color: [text_color];
+	"}
+
+/obj/screen/text/blurb/proc/hide_from(list/client/viewers)
+	PRIVATE_PROC(TRUE)
+
+	fade()
+	addtimer(CALLBACK(src, TYPE_PROC_REF(/obj/screen/text/blurb, remove_from_viewers), viewers), fade_animation_duration)
+
+/obj/screen/text/blurb/proc/appear()
+	PRIVATE_PROC(TRUE)
+
+	animate(src, alpha = 255, time = appear_animation_duration)
+
+
+/obj/screen/text/blurb/proc/fade()
+	PRIVATE_PROC(TRUE)
+
+	animate(src, alpha = 0, time = fade_animation_duration)
+
+
+/obj/screen/text/blurb/proc/print_text()
+	PRIVATE_PROC(TRUE)
```

Comment:
You should probably check for client here if you're gonna be sleeping, just in case they logout immediately on roundstart.

Reply:
Client is checked with the single person showed to, used on round start / spawning in.
Server wide round is restarting is set to everyone on the server.

Reply:
which also only shows to people with active clients